urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/ReactAirshipPreferences.java

```diff
@@ -218,15 +218,19 @@ private SharedPreferences.Editor edit() {
     }
 
     private void ensurePreferences() {
-        if (!created.getAndSet(true)) {
-            this.preferences = context.getSharedPreferences(SHARED_PREFERENCES_FILE, Context.MODE_PRIVATE);
-
-            // Migrate any data stored in default
-            SharedPreferences defaultPreferences = PreferenceManager.getDefaultSharedPreferences(context);
-            if (defaultPreferences.contains(AUTO_LAUNCH_MESSAGE_CENTER)) {
-                boolean autoLaunchMessageCenter = defaultPreferences.getBoolean(AUTO_LAUNCH_MESSAGE_CENTER, true);
-                defaultPreferences.edit().remove(AUTO_LAUNCH_MESSAGE_CENTER).apply();
-                this.preferences.edit().putBoolean(AUTO_LAUNCH_MESSAGE_CENTER, autoLaunchMessageCenter).apply();
+        synchronized (this) {
+            if (preferences == null) {
+                this.preferences = context.getSharedPreferences(SHARED_PREFERENCES_FILE, Context.MODE_PRIVATE);
+            }
+
+            if (!created.getAndSet(true)) {
```

Comment:
you can remove this and just move the migration logic into `preferences  == null` check